### ISSUE TITLE
Support custom attribution strings

### ISSUE THREAD

**Issue by AlanPew** [mapbox/mapbox-gl-js#1485]
I'm having trouble figuring out how to add a custom attribution to my maps in addition to the standard attributions already added. When I call..... map.addControl(new mapboxgl.Attribution()); where do I add a custom string?  Probably a stupid question, but just trying to do the right thing with attributions before I share the awesomeness that is customized Mapbox GL maps.

**Comment by jfirebaugh** [comment]
Good question. Unfortunately there's not a good way to add a custom attribution right now. We'll look into adding this.

**Comment by lyzidiamond** [comment]
tagging @mapbox/support for known issue

**Comment by nilshoerrmann** [comment]
I just encountered this issue as well. Would it be possible to add an additional `attribution` key to the different source objects that let us specify custom attributions? For example:

```javascript
"geojson-marker": {
  "type": "geojson",
  "data": { … },
  "attribution": "My custom attribution text."
}
```

Or:

```javascript
var sourceObj = new mapboxgl.GeoJSONSource({
   data: { … },
   attribution: "My custom attribution text."
});
```

**Comment by stepankuzmin** [comment]
I've found custom attribution strings support for `vector` sources [here](https://github.com/mapbox/mapbox-gl-js/blob/master/test/js/ui/control/attribution.test.js#L82-L87), but it seems that there is no support for `geojson` sources according to [this discussion](https://github.com/mapbox/mapbox-gl-style-spec/issues/380).

**Comment by andrewharvey** [comment]
I've just noticed that, at least for raster sources you can just pass an attribution property to the source options https://www.mapbox.com/mapbox-gl-js/style-spec/#sources-raster (although it's not actually documented there).

**Comment by andrewharvey** [comment]
Until this is implemented, one workaround is to set the attribution string directly:

```js
map.addLayer({
    "id": "attribution-layer",
    "type": "circle",
    "source": {
        "type": "geojson",
        "data": {
            "type": "Feature",
            "properties": {},
            "geometry": null
        }
    }
});
map.style.sourceCaches['attribution-layer']._source.attribution = "&copy; foobar";
```

**Comment by 1ec5** [comment]
The equivalent issue for iOS is tracked in mapbox/mapbox-gl-native#9024.

**Comment by pathmapper** [comment]
It looks like this issue was closed unintended by #6364 because of the word _close_ in the PR description.

> I think to close #1485 we would still need to allow custom attribution strings passed directly to the AttributionControl not attached to any source.

Although there are workarounds available, custom attribution strings are not implemented yet.

**Comment by andrewharvey** [comment]
@pathmapper You're right passing a fixed string directly to the AttributionControl isn't yet supported, but one workaround is pretty straight forward now:

```js
map.addSource('attribution', {
   type: 'geojson',
   data: {
      type: 'FeatureCollection',
      features: []
   },
   attribution: 'Your custom attribution'
});
```
(until #6281, then you'll need to add a layer too)

**Comment by andrewharvey** [comment]
closed by #7033

**Comment by jfoshee** [comment]
Docs: https://docs.mapbox.com/mapbox-gl-js/api/markers/#attributioncontrol
